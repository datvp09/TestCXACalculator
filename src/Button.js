import React from 'react';
import {View, TouchableHighlight, Text, StyleSheet} from 'react-native';

const Button = React.memo((props) => (
  <TouchableHighlight
    style={[styles.buttonStyles]}
    onPress={props.onPress}
    underlayColor="#777">
    <View style={[styles.contButton, props.style]}>
      <Text style={[styles.txtButtons, props.textStyle]}>{props.text}</Text>
    </View>
  </TouchableHighlight>
));

const styles = StyleSheet.create({
  txtButtons: {
    color: '#000',
    fontFamily: 'System',
    fontSize: 40,
    fontWeight: '500',
  },
  buttonStyles: {
    flex: 1,
  },
  contButton: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: 'red',
  },
});

export default Button;
