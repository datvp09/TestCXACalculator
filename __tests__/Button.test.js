import 'react-native';
import React from 'react';
import Button from '../src/Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = renderer
        .create(
          <Button
            text={'9'}
            onPress={() => console.log('create-9')}
            style={{}}
          />,
        )
        .toJSON();
      expect(component).toMatchSnapshot();
    });
  });
});
