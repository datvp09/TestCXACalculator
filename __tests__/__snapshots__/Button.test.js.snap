// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "red",
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 40,
            "fontWeight": "500",
          },
          undefined,
        ]
      }
    >
      9
    </Text>
  </View>
</View>
`;
